### PR TITLE
Replace almost every use of browser with endpoint

### PR DESCRIPTION
**Pull request by ekr**

### CODE REVIEW

**Comment by juberti** [comment]
Should we treat _endpoint_ and _implementation_ as interchangeable?

File: draft-ietf-rtcweb-jsep.xml [hunk 1]
```diff
@@ -93,17 +93,20 @@ ipr="trust200902">
         <t>With these considerations in mind, this document describes
         the Javascript Session Establishment Protocol (JSEP) that
         allows for full control of the signaling state machine from
-        Javascript. JSEP removes the browser almost entirely from the
-        core signaling flow, which is instead handled by the Javascript
-        making use of two interfaces: (1) passing in local and remote
+        Javascript. As described above, JSEP assumes a model in which
+        a Javascript application runs inside a WebRTC enabled runtime
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: WebRTC-enabled

File: draft-ietf-rtcweb-jsep.xml [hunk 2]
```diff
@@ -93,17 +93,20 @@ ipr="trust200902">
...
+        (the "JSEP endpoint" or "endpoint"). JSEP removes the endpoint
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: endpoint -> runtime

Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: This doesn't seem quite right. We want to say that the WebRTC 'guts' are removed from the signaling flow, but I'm not sure that's properly captured by the use of 'endpoint'.

Stated another way: does a "JSEP endpoint" include the application code? It sounds like the answer is no, but that's kind of unintuitive.

**Comment by ekr** [comment]
@jubert: here is the policy I followed

- JSEP implementation (or occasionally "implementation" when I just used JSEP right before) is the term for the browser
- JSEP endpoint is the term for the app/browser combo.

PTAL

File: draft-ietf-rtcweb-jsep.xml [hunk 3]
```diff
@@ -93,17 +93,21 @@ ipr="trust200902">
         <t>With these considerations in mind, this document describes
         the Javascript Session Establishment Protocol (JSEP) that
         allows for full control of the signaling state machine from
-        Javascript. JSEP removes the browser almost entirely from the
-        core signaling flow, which is instead handled by the Javascript
-        making use of two interfaces: (1) passing in local and remote
+        Javascript. As described above, JSEP assumes a model in which
+        a Javascript application runs inside a runtime containing
+        WebRTC APIs (the "JSEP implementation"). The JSEP implementation
+        almost entirely divorced from the core signaling flow, which is instead
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: 'is almost'

File: draft-ietf-rtcweb-jsep.xml [hunk 4]
```diff
@@ -93,17 +93,21 @@ ipr="trust200902">
         <t>With these considerations in mind, this document describes
         the Javascript Session Establishment Protocol (JSEP) that
         allows for full control of the signaling state machine from
-        Javascript. JSEP removes the browser almost entirely from the
-        core signaling flow, which is instead handled by the Javascript
-        making use of two interfaces: (1) passing in local and remote
+        Javascript. As described above, JSEP assumes a model in which
+        a Javascript application runs inside a runtime containing
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: runs -> executes?

File: draft-ietf-rtcweb-jsep.xml [hunk 5]
```diff
@@ -93,17 +93,21 @@ ipr="trust200902">
...
+        WebRTC APIs (the "JSEP implementation"). The JSEP implementation
+        almost entirely divorced from the core signaling flow, which is instead
+        handled by the Javascript making use of two interfaces: (1) passing in local and remote
         session descriptions and (2) interacting with the ICE state
-        machine.</t>
+        machine. The combination of the JSEP implementation and the Javascript
+        application is referred to throughout this document as a "JSEP endpoint." </t>
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: Period after "

File: draft-ietf-rtcweb-jsep.xml [hunk 6]
```diff
@@ -93,17 +93,21 @@ ipr="trust200902">
...
 
         <t>In this document, the use of JSEP is described as if it
-        always occurs between two browsers. Note though in many cases
-        it will actually be between a browser and some kind of server,
-        such as a gateway or MCU. This distinction is invisible to the
-        browser; it just follows the instructions it is given via the
+        always occurs between two JSEP endpoints. Note though in many cases
+        it will actually be between a JSEP endpoint and some kind of
+        server, such as a gateway or MCU. As long as the server
+        generates JSEP-compliant SDP, this distinction is invisible to the
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: Is this JSEP-compliant SDP comment accurate? We will handle SDP even that doesn't conform to what a JSEP implementation would generate, as long as it meets our mandatory-to-use statutes.

File: draft-ietf-rtcweb-jsep.xml [hunk 7]
```diff
@@ -127,8 +131,8 @@ ipr="trust200902">
         <t>Regarding ICE
         <xref target="RFC5245"></xref>, JSEP decouples the ICE state
         machine from the overall signaling state machine, as the ICE
-        state machine must remain in the browser, because only the
-        browser has the necessary knowledge of candidates and other
+        state machine must remain in the JSEP implementation, because only the
+        JSEP implementation has the necessary knowledge of candidates and other
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: JSEP implementation -> implementation (no need to repeat the whole thing)

File: draft-ietf-rtcweb-jsep.xml [hunk 8]
```diff
@@ -189,9 +193,9 @@ ipr="trust200902">
 
         <t>One variation on JSEP that was considered was to keep the
         basic session description-oriented API, but to move the
-        mechanism for generating offers and answers out of the browser.
+        mechanism for generating offers and answers out of the JSEP implementation,
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: , -> .

File: draft-ietf-rtcweb-jsep.xml [hunk 9]
```diff
@@ -189,9 +193,9 @@ ipr="trust200902">
...
         Instead of providing createOffer/createAnswer methods within
-        the browser, this approach would instead expose a
+        the JSEP implementation, this approach would instead expose a
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: JSEP implementation -> implementation

File: draft-ietf-rtcweb-jsep.xml [hunk 10]
```diff
@@ -204,7 +208,7 @@ ipr="trust200902">
         Providing createOffer/createAnswer avoids this problem, but
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: I think this paragraph is no longer accurate.

File: draft-ietf-rtcweb-jsep.xml [hunk 11]
```diff
@@ -226,13 +230,13 @@ ipr="trust200902">
...
         provides mechanisms to create offers and answers, as well as to
-        apply them to a session. However, the browser is totally
+        apply them to a session. However, the JSEP implementation is totally
         decoupled from the actual mechanism by which these offers and
         answers are communicated to the remote side, including
         addressing, retransmission, forking, and glare handling. These
         issues are left entirely up to the application; the application
         has complete control over which offers and answers get handed
-        to the browser, and when.</t>
+        to the JSEP implementation, and when.</t>
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: JSEP implementation -> implementation

File: draft-ietf-rtcweb-jsep.xml [hunk 12]
```diff
@@ -606,7 +611,7 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
           comes with corresponding operational costs). To accomplish this,
           JSEP allows the application to restrict which ICE candidates
           are used in a session. Note that this
-          filtering is applied on top of any restrictions the browser
+          filtering is applied on top of any restrictions the JSEP implementation
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: JSEP implementation -> implementation

File: draft-ietf-rtcweb-jsep.xml [hunk 13]
```diff
@@ -618,13 +623,13 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
-          browser MUST allow the candidate policy to be changed in
+          JSEP implementation MUST allow the candidate policy to be changed in
           mid-session, subject to the aforementioned interactions with
           local policy.</t>
 
-          <t>To administer the ICE candidate policy, the browser will
+          <t>To administer the ICE candidate policy, the JSEP implementation will
           determine the current setting at the start of each gathering
-          phase. Then, during the gathering phase, the browser MUST NOT
+          phase. Then, during the gathering phase, the JSEP implementation MUST NOT
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: JSEP implementation -> implementation

File: draft-ietf-rtcweb-jsep.xml [hunk 14]
```diff
@@ -636,18 +641,18 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
 
-          <t>JSEP applications typically inform the browser to begin
+          <t>JSEP applications typically inform the JSEP implementation to begin
           ICE gathering via the information supplied to
           setLocalDescription, as this is where the app specifies the
           number of media streams, and thereby ICE components, for
           which to gather candidates. However, to accelerate cases
           where the application knows the number of ICE components to
-          use ahead of time, it may ask the browser to gather a pool of
+          use ahead of time, it may ask the endpoint to gather a pool of
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: is endpoint the right term here? I expected 'implementation'.

File: draft-ietf-rtcweb-jsep.xml [hunk 15]
```diff
@@ -996,13 +1001,13 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
-          browser to only surface the permitted candidates (including
-          any internal browser filtering) to the application, and only
+          JSEP implementation to only surface the permitted candidates (including
+          any implementation-internal filtering) to the application, and only
           use those candidates for connectivity checks. The set of
           available policies is as follows:
           <list style="hanging">
-            <t hangText="all:">All candidates permitted by browser
-            policy will be gathered and used.</t>
+            <t hangText="all:">All candidates permitted by JSEP
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: JSEP implementation -> implementation

File: draft-ietf-rtcweb-jsep.xml [hunk 16]
```diff
@@ -1854,11 +1859,11 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
         <section title="Profile Names and Interoperability"
         anchor="sec.profile-names">
 
-          <t>For media m= sections, JSEP endpoints MUST support both
+          <t>For media m= sections, JSEP implementations MUST support both
           the "UDP/TLS/ RTP/SAVPF" and "TCP/DTLS/RTP/SAVPF" profiles
           and MUST indicate one of these two profiles for each media m=
           line they produce in an offer. For data m= sections, JSEP
-          endpoints must support both the "UDP/DTLS/SCTP" and
+          implementations must support both the "UDP/DTLS/SCTP" and
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: JSEP implementations -> implementations

File: draft-ietf-rtcweb-jsep.xml [hunk 17]
```diff
@@ -1897,13 +1902,13 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
               that AVPF is being used in an AVP compatible mode and
               use a value of "trr-int=4000".</t>
 
-              <t>For data m= sections, JSEP endpoints MUST support
+              <t>For data m= sections, JSEP implementations MUST support
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: JSEP implementations -> implementations

**Comment by ekr** [comment]
@juberi I think this is on you.

**Pull request merged**